src/ui/marker.js

```diff
@@ -277,9 +279,26 @@ export default class Marker extends Evented {
      * @returns {Marker} `this`
      */
     setPopup(popup: ?Popup) {
+        const keypressListener = (e: KeyboardEvent) => {
+            const code = e.code;
+            const legacyCode = e.charCode || e.keyCode;
+
+            if (
+                (code === 'Space') || (code === 'Enter') ||
+                (legacyCode === 32) || (legacyCode === 13) // space or enter
+            ) {
+                this.togglePopup();
+            }
+        };
```

Comment:
For consistency, can we move this into its own `_onKeyPress` method (and add it to `bindAll` in the constructor)?

Reply:
I could, but for reasons I don't understand I need to do:

```js
this._element.addEventListener('keypress', (e: KeyboardEvent) => { this._onKeyPress(e) });
```

as the following doesn't work:

```js
this._element.addEventListener('keypress', this._onKeyPress);
```

is that okay?

Reply:
Actually, forget that, I must have been doing something wrong, it's working now.